File: app/src/main/java/com/bank/testeandroidv2/loginScreen/LoginModel.java
package com.bank.testeandroidv2.loginScreen;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

public class LoginModel {
    String user;
    String password;
    int status;
    @SerializedName("userAccount")
    @Expose
    private UserAccount userAccount;

    @SerializedName("error")
    @Expose
    private Object error;

    public UserAccount getUserAccount() {
        return userAccount;
    }

    public void setUserAccount(UserAccount userAccount) {
        this.userAccount = userAccount;
    }

    public Object getError() {
        return error;
    }

    public void setError(Object error) {
        this.error = error;
    }
}

class LoginViewModel {
    int status;
    UserAccount userAccount;
    Object error;
}

class LoginRequest {
    String user;
    String password;
}

class LoginResponse {
    int status;
    UserAccount userAccount;
    Object error;
}

class UserAccount {
    Long userId;
    String name;
    String bankAccount;
    String agency;
    Double balance;
}

File: app/src/main/java/com/accenture/primo/bankapp/ui/login/LoginModel.kt
package com.accenture.primo.bankapp.ui.login

import com.accenture.primo.bankapp.model.Error
import com.accenture.primo.bankapp.model.User
import com.google.gson.annotations.SerializedName

class LoginModel
{
    class LoginViewModel(val user: User, val login: LoginRequest)

    class LoginRequest(val user: String, val password: String)

    class LoginResponse(
        @SerializedName(value = "userAccount")
        val user: User,
        @SerializedName(value = "error")
        val error: Error
    )
}


